### PR TITLE
Added ShoppingCart Service, Dao and Entity; added Ticket Dao and Entity

### PR DESCRIPTION
**Pull request by aleksandr-hrankin**

### CODE REVIEW

File: src/main/java/ua/mate/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
+
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
```
Comment (dariiabutorina) [inline src/main/java/ua/mate/cinema/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        ticketDao.add(ticket);
        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
```

File: src/main/java/ua/mate/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession().getSession()) {
+            Query<ShoppingCart> getShoppingCartByUserQuery
+                    = session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "where sc.user = :user", ShoppingCart.class);
+            getShoppingCartByUserQuery.setParameter("user", user);
+            return getShoppingCartByUserQuery.getSingleResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get shopping cart by user " + user, e);
+        }
```
Comment (dariiabutorina) [inline src/main/java/ua/mate/cinema/dao/impl/ShoppingCartDaoImpl.java]: No need to catch an exception here
If we've got the user registered, the shopping cart definitely exists

**Review by dariiabutorina — COMMENTED** [review]
Good job :)

File: src/main/java/ua/mate/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession().getSession()) {
+            Query<ShoppingCart> getShoppingCartByUserQuery
+                    = session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "where sc.user = :user", ShoppingCart.class);
```
Comment (dmytro-hryhoruk) [inline src/main/java/ua/mate/cinema/dao/impl/ShoppingCartDaoImpl.java:43]: will you have access to this shoppingCart user after session is over?

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**